packages/api/cms-api/src/auth/services/jwt.auth-service.ts

```diff
@@ -1,27 +1,36 @@
 import { Injectable, Type } from "@nestjs/common";
+import { ModuleRef } from "@nestjs/core";
 import { JwtService, JwtVerifyOptions } from "@nestjs/jwt";
 import { Request } from "express";
 import JwksRsa, { JwksClient } from "jwks-rsa";
 
 import { User } from "../../user-permissions/interfaces/user";
 import { AuthenticateUserResult, AuthServiceInterface, SKIP_AUTH_SERVICE } from "../util/auth-service.interface";
 
-type JwtPayload = { [key: string]: unknown };
+// eslint-disable-next-line @typescript-eslint/no-explicit-any
+export type JwtPayload = { [key: string]: any };
+
+export interface JwtToUserServiceInterface {
+    convertJwtToUser: (jwt: JwtPayload) => Promise<User> | User;
+}
 
 export interface JwtAuthServiceOptions {
     jwksOptions?: JwksRsa.Options;
     verifyOptions?: JwtVerifyOptions;
     tokenHeaderName?: string;
-    // eslint-disable-next-line @typescript-eslint/no-explicit-any
-    convertJwtToUser?: (jwt: any) => Promise<User> | User;
+    convertJwtToUser?: (jwt: JwtPayload) => Promise<User> | User;
+    convertJwtToUserService?: Type<JwtToUserServiceInterface>;
```

Comment:
In other use cases we use a single option and determine if its a service during runtime, e.g., [here](https://github.com/vivid-planet/comet/blob/main/packages/api/cms-api/src/user-permissions/content-scope.service.ts#L92-L97). Should we do this here as well?

Reply:
I'd use a `convertJwtToUser?: ((jwt: JwtPayload) => Promise<User> | User) | Type<JwtToUserServiceInterface>;`, because we have similar support (service OR function) in other apis. Decorators only though.

(It is unusual that you have to get the service instance from ModuleRef (normally dependencies get constructor-injected) - but that's because you have a factory)